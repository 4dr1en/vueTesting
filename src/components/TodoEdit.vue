<script setup lang="ts">
import { ref } from "vue";

const props = defineProps<{
  oldTodo: string;
}>();

const todo = ref<string>(props.oldTodo);

defineEmits<{
  (e: "editTodo", todo: string): void;
}>();
</script>

<template>
  <div>
    <label for="todo">Todo : </label>
    <input
      type="text"
      v-model="todo"
      @keydown.enter="$emit('editTodo', todo)"
      id="todo"
    />
    <button @click="$emit('editTodo', todo)">Update</button>
  </div>
</template>

<style lang="scss"></style>
