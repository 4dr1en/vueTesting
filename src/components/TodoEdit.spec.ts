import TodoEdit from "./TodoEdit.vue";
import { mount } from "@vue/test-utils";
import { expect, describe, it } from "vitest";

describe("TodoEdit", () => {
  it("Prepend the input field withe the old value", async () => {
    const wrapper = mount(TodoEdit, {
      props: {
        oldTodo: "Buy groceries",
      },
    });
    const input = wrapper.find("input");
    expect(input.element.value).toBe("Buy groceries");
  });

  it('Emits a "editTodo" event when the form is submitted', async () => {
    const wrapper = mount(TodoEdit, {
      props: {
        oldTodo: "Buy groceries",
      },
    });
    const input = wrapper.find("input");
    const newValue = "Go to the gym";
    await input.setValue(newValue);
    const button = wrapper.find("button");
    await button.trigger("click");

    expect(wrapper.emitted("editTodo")).toHaveLength(1);
    expect(wrapper.emitted("editTodo")![0]![0]).toBe(newValue);
  });

  it("Should match the snapshot", () => {
    const wrapper = mount(TodoEdit, {
      props: {
        oldTodo: "Buy groceries",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
