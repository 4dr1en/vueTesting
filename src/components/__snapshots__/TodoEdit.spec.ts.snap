// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoEdit > Should match the snapshot 1`] = `"<div><label for=\\"todo\\">Todo : </label><input type=\\"text\\" id=\\"todo\\"><button>Update</button></div>"`;
